src/scmdata/ops.py

```diff
@@ -566,14 +620,52 @@ def integrate(self, out_var=None):
     out_var : str
         If provided, the variable column of the output is set equal to
         ``out_var``. Otherwise, the output variables are equal to the input
-        variables, prefixed with "Cumulative " .
+        variables, prefixed with "Cumulative ".
+
+    out_unit : str
+        If provided, the output will be converted to the target unit. The output
+        will have a different dimensionality than the input data so `out_unit`
+        should reflect this.
+
+    method : str
+        Method used to integrate. The selected methods have pros/cons depending
+        on the intended use-case.
+
+        The available options are:
+        * 'sum' - calculate the cumulative sum of each timeseries. This
+         method is best used to integrate timeseries of emissions (or
+         other piecewise constant timeseries) when the data are on annual time
+         steps. This method requires timeseries on a uniform, annual interval.
+        * 'trapz' (default) - uses the trapezoid rule to approximate the integral
+         of a timeseries. This function should be used for piecewise-linear
+         timeseries, i.e. concentrations or decadal estimates of emissions.
+         This method handles non-uniform intervals without having to resample to
+          annual values.
+
+        If you wish to calculate cumulative emissions, the preferred approach
+        should be to resample to annual values and then use the `sum` method.
+
+    reference_year: int
```

Comment:
I would suggest leaving this out, people can always call `relative_to_ref_period_mean` after they've called `integrate` no?

Reply:
I don't think it is the same thing and we should be very deliberate about it. With the sum method you end up not capturing the first year

Reply:
Well if the convention is 'up to and including' then you just need to call `relative_to_ref_period_mean` with the previous year to make it work (e.g. if you want cumulative emissions rel. to 2020 then you have to call `.integrate().relative_to_ref_period_mean(year=2020)`. Probably having a `reference_year` argument here makes things simpler I guess.

I think the other problem we have here is that emissions are piecewise constant while cumulative emissions are piecewise linear but we have no way to point that out to the user at the moment (e.g. if you calculate cumulative emissions off annual data, the outputs should all be on Dec 31 of the year and then you signal that the result are 'end of year' results i.e. each year value is 'up to and including'). I'm pretty sure this is the rabbit hole we went down with scmdata but as we know we never find a really satisfactory solution.

Reply:
You're spot on. We are falling down that same hole - this function consumes two different types of timeseries which look the same but act differently. Perhaps that suggests that we expose `cumtrapz` and `cumsum` and leave it up to the users?

Reply:
I'm not sure if we can converge the conventions for trapz and sum into one set

Reply:
> Perhaps that suggests that we expose cumtrapz and cumsum and leave it up to the users?

That's probably smartest.

> I'm not sure if we can converge the conventions for trapz and sum into one set

Would our convention for cumsum be 'up to and including' but not for cumtrapz then?

Reply:
>Would our convention for cumsum be 'up to and including' but not for cumtrapz then?
Up to and including the target year for `cumsum` (the concept of days and months doesn't exist). `cumtrapz` is up to the point in time